internal/pkg/warewulfd/parser.go

```diff
@@ -25,16 +27,22 @@ func parseReq(req *http.Request) (parserInfo, error) {
 	url := strings.Split(req.URL.Path, "?")[0]
 	path_parts := strings.Split(url, "/")
 
-	if len(path_parts) != 3 {
+	if len(path_parts) < 3 {
 		return ret, errors.New("unknown path components in GET")
 	}
 
 	// handle when stage was passed in the url path /[stage]/hwaddr
 	stage := path_parts[1]
-	hwaddr := path_parts[2]
-	hwaddr = strings.ReplaceAll(hwaddr, "-", ":")
-	hwaddr = strings.ToLower(hwaddr)
-
+	hwaddr := ""
+	if stage != "efiboot" {
+		hwaddr = path_parts[2]
+		hwaddr = strings.ReplaceAll(hwaddr, "-", ":")
+		hwaddr = strings.ToLower(hwaddr)
+	} else if len(path_parts) > 3 {
+		ret.efifile = strings.Join(path_parts[2:], "/")
+	} else {
+		ret.efifile = path_parts[2]
+	}
```

Comment:
What if path_parts > 3 and stage == efiboot?

At least, "unknown path components in GET" probably needs to be adjusted. (It was probably inaccurate before, too.)